quickwit/quickwit-config/src/index_config/serialize.rs

```diff
@@ -64,6 +64,50 @@ pub fn load_index_config_from_user_config(
     index_config_for_serialization.build_and_validate(Some(default_index_root_uri))
 }
 
+/// Parses and validates an [`IndexConfig`] update. Ensures that the new
+/// configuration is valid in itself and compared to the current index config.
+/// If the new configuration omits some fields, the default values will be used,
+/// not those of the current index config. The only exception is the index_uri
+/// because it cannot be updated.
```

Comment:
I am unsure it is wise to special case `index_uri` here and not apply the default when it is not present in the update config.